### PR TITLE
Implement signature verification for spawn_npc

### LINKED ISSUE #31: feat: spawn_npc should check signature
`spawn_npc` should check that the message sent was correctly signed by lore-machine

### PR DESCRIPTION
**Pull request by cwastche**
Implement signature verification for spawn_npc
Add pub_key to NpcConfig and its setter.
Closes #31

### CODE REVIEW

File: client/src/components/cityview/realm/villagers/NpcPanel.tsx [hunk 1]
```diff
@@ -63,6 +64,9 @@ export const NpcPanel = ({ type = "all" }: NpcPanelProps) => {
     setSpawned,
   } = useNpcContext();
 
+  console.log(npcs);
```
Comment (edisontim) [inline client/src/components/cityview/realm/villagers/NpcPanel.tsx:67]: remove logs

File: contracts/src/systems/npc/utils.cairo [hunk 2]
```diff
@@ -29,3 +31,21 @@ fn pedersen_hash_many(data: Span<felt252>) -> felt252 {
     };
     pedersen(current_hash, data.len().into())
 }
+
+fn format_data_to_hash(nonce: felt252, characteristics: felt252, character_trait: felt252, full_name: felt252) ->Span<felt252> {
```
Comment (edisontim) [inline contracts/src/systems/npc/utils.cairo:35]: Could be a bit more explicit with the naming in this function for the function name and the data_array

File: client/src/components/cityview/realm/villagers/NpcPanel.tsx (hunk 1 above)
Comment (cwastche) [reply client/src/components/cityview/realm/villagers/NpcPanel.tsx]: Done.

File: contracts/src/systems/npc/utils.cairo (hunk 2 above)
Comment (cwastche) [reply contracts/src/systems/npc/utils.cairo]: Done.

**Review by edisontim — APPROVED** [review]
lgtm :)

**Pull request merged**